File: src/com/l2kt/gameserver/network/serverpackets/SkillCoolTime.java
package com.l2kt.gameserver.network.serverpackets;

import java.util.List;
import java.util.stream.Collectors;

import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.model.holder.Timestamp;

public class SkillCoolTime extends L2GameServerPacket
{
	public List<Timestamp> _reuseTimeStamps;
	
	public SkillCoolTime(Player cha)
	{
		_reuseTimeStamps = cha.getReuseTimeStamps().stream().filter(r -> r.hasNotPassed()).collect(Collectors.toList());
	}
	
	@Override
	protected void writeImpl()
	{
		writeC(0xc1);
		writeD(_reuseTimeStamps.size()); // list size
		for (Timestamp ts : _reuseTimeStamps)
		{
			writeD(ts.getId());
			writeD(ts.getValue());
			writeD((int) ts.getReuse() / 1000);
			writeD((int) ts.getRemaining() / 1000);
		}
	}
}

File: src/com/l2kt/gameserver/network/serverpackets/SkillCoolTime.kt
package com.l2kt.gameserver.network.serverpackets

import com.l2kt.gameserver.model.actor.instance.Player
import com.l2kt.gameserver.model.holder.Timestamp

class SkillCoolTime(cha: Player) : L2GameServerPacket() {
    private var _reuseTimeStamps: List<Timestamp> = cha.reuseTimeStamps.filter { r -> r.hasNotPassed() }

    override fun writeImpl() {
        writeC(0xc1)
        writeD(_reuseTimeStamps.size) // list size
        for (ts in _reuseTimeStamps) {
            writeD(ts.id)
            writeD(ts.value)
            writeD(ts.reuse.toInt() / 1000)
            writeD(ts.remaining.toInt() / 1000)
        }
    }
}